lib/interpreter.ml

```diff
@@ -82,19 +86,42 @@ module Make
             let/ state = Choice.get in
             Store.set state.store x symb_val;
             return cont)
-    | Symbol_int_c (x, s, c) -> (
-        let symb_opt = make_symbol s "int" in
-        match symb_opt with
-        | None ->
-            failwith
-              "ApplicationError: tried to create a symbolic value in a \
-               concrete execution context"
-        | Some symb_val ->
+    | Symbol_int_c (x, s, c) -> 
+      let symb_opt = make_symbol s "int" in
+      (match symb_opt with
+      | None ->
+        failwith
+          "ApplicationError: tried to create a symbolic value in a \
+            concrete execution context"
+      | Some symb_val ->
+        let f_symb_int (s : state) =
+          Store.set s.store x symb_val;
+          let v = eval s.store c in
+          let pc' = add_condition s.pc v in
+          if is_true pc' 
+            then [ ((), SState.{ s with pc = pc' })]
+            else [] in 
+        let/ _ = Choice.lift f_symb_int in 
+        return cont)
+
+      (*
+      (
+            fun 
             let/ state = Choice.get in
             Store.set state.store x symb_val;
             let e = eval state.store c in
             let pc' = add_condition state.pc e in
             if is_true pc' then return cont else Choice.return AssumeF)
+        *)
+    
+    (*
+      let/ state = Choice.get in
+        let v = eval state.store e in
+        let/ b = Choice.select v in
+        if b then return cont else Choice.return Outcome.AssumeF   
+    *)
+
+
```

Comment:
Can we remove these comments? Otherwise, we probably should describe why they are here instead.